geomstats/geometry/spd_matrices_space.py

```diff
@@ -246,19 +246,23 @@ def __init__(self, n, power_affine=1):
                 dimension=dimension,
                 signature=(dimension, 0, 0))
         self.n = n
+        self.power_affine = power_affine
 
     def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
         """
         Compute the inner product of tangent_vec_a and tangent_vec_b
         at point base_point using the affine invariant Riemannian metric.
         """
+        power_affine = self.power_affine
         tangent_vec_a = gs.to_ndarray(tangent_vec_a, to_ndim=3)
         n_tangent_vecs_a, _, _ = tangent_vec_a.shape
         tangent_vec_b = gs.to_ndarray(tangent_vec_b, to_ndim=3)
         n_tangent_vecs_b, _, _ = tangent_vec_b.shape
 
         base_point = gs.to_ndarray(base_point, to_ndim=3)
-        n_base_points, _, _ = base_point.shape
+        n_base_points, dim, _ = base_point.shape
+
+        spd_space = SPDMatricesSpace(dim)
```

Comment:
You don't need this, you can call `self`.

Reply:
Aaah right! Sorry, I read too fast. The question then is: should we have `space` as an attribute of each SPDMetric?